compiler/ilgen/OMRIlBuilder.cpp

```diff
@@ -1005,6 +1005,31 @@ OMR::IlBuilder::convertTo(TR::DataType typeTo, TR::IlValue *v, bool needUnsigned
    return convertedValue;
    }
 
+static TR::ILOpCodes
+getCoercionOpcode(TR::DataType fromType, TR::DataType toType)
+   {
+   if (fromType == TR::Int32 && toType == TR::Float) return TR::ibits2f;
+   else if (fromType == TR::Float && toType == TR::Int32) return TR::fbits2i;
+   else if (fromType == TR::Int64 && toType == TR::Double) return TR::lbits2d;
+   else if (fromType == TR::Double && toType == TR::Int64) return TR::dbits2l;
+   else return TR::BadILOp;
+   }
+
+TR::IlValue*
+OMR::IlBuilder::CoerceTo(TR::IlType* t, TR::IlValue* v)
+   {
+   TR::DataType typeFrom = v->getDataType();
+   TR::DataType typeTo = t->getPrimitiveType();
+
+   TR::ILOpCodes coerceOpcode = getCoercionOpcode(typeFrom, typeTo);
```

Comment:
should insert an `if (toType != fromType)` around all this code...if the type is already correct, then you'll get a BadILOp, but there's no real reason to assert in that scenario.

Reply:
Fixed!